### ISSUE TITLE
Bug in TLS Server Automaton: buffer_in is not emptied between connections

### ISSUE THREAD

**Issue by pictyeye** [secdev/scapy#2129]
When a client sends a parsable but incorrect ClientKeyExchange, the TLS server automaton switches to the MISSING_CLIENTKEYEXCHANGE state and eventually returns to the state accepting client connections. However, the invalid record is still in buffer_in, which makes the server completely unusable.

Using the current master branch (a1288aab92972e6c353cfb0b736f04c6ad11d1d6), the simple client attached triggers the bug (it sends a ClientHello and a parsable but wrong ClientKeyExchange, then opens another connection, only to be greeted by an unsollicited alert).

[scapy-bug-buffer_in.py](https://github.com/secdev/scapy/files/3359603/scapy-bug-buffer_in.txt)

**Comment by guedou** [comment]
Thanks for reporting and fixing the issue. I am glad to see that you are still active on the TLS side =)

FTR, here is how to reproduce:
```
sudo ./run_scapy
>>> load_layer("tls")
>>> t = TLSServerAutomaton(mycert='test/tls/pki/srv_cert.pem', mykey='test/tls/pki/srv_key.pem')
>>> t.run()
```

In another terminal: `python scapy-bug-buffer_in.py`

**Comment by guedou** [comment]
Fixed by #2130
